Protecting users from seeing other users information

VictorKostiuk opened the pull request.

VictorKostiuk committed:
Push to Heroku

VictorKostiuk committed:
Ruby version was updated to 3.0.4

VictorKostiuk committed:
removed belongs_to total_sum in checks migration

VictorKostiuk committed:
fixed transaction

VictorKostiuk committed:
Changed file config/database.yml

VictorKostiuk committed:
Changed config

VictorKostiuk committed:
fix migration

VictorKostiuk committed:
bundle update

VictorKostiuk committed:
Changed database configuration

VictorKostiuk committed:
Changed configuration

VictorKostiuk committed:
Changed migrations

VictorKostiuk committed:
Added migration AddTransactionsToAccount

VictorKostiuk committed:
protect users from seeing other users information

orest-kostiuk commented on .idea/MoneyBudget.iml:
```diff
@@ -26,114 +26,114 @@
       <excludeFolder url="file://$MODULE_DIR$/public/system" />
       <excludeFolder url="file://$MODULE_DIR$/components" />
     </content>
-    <orderEntry type="jdk" jdkName="RVM: ruby-3.0.0" jdkType="RUBY_SDK" />
```
Please, remove it from git

orest-kostiuk commented on .idea/inspectionProfiles/Project_Default.xml:
```diff
@@ -0,0 +1,6 @@
+<component name="InspectionProjectProfileManager">
```
Please, remove it from git

orest-kostiuk commented on Gemfile:
```diff
@@ -15,8 +15,7 @@ gem 'cssbundling-rails'
 # Use sqlite3 as the database for Active Record
-gem 'sqlite3', '~> 1.4'
-
+# gem 'sqlite3', '~> 1.4'
```
remove it

orest-kostiuk commented on app/controllers/accounts_controller.rb:
```diff
@@ -42,7 +42,7 @@ def destroy
 
   def total_sum_params
-    params.require(:account).permit(:title, :for_what, :total_count)
+    params.require(:account).permit(:title, :for_what, :total_count).merge(user_id: current_user.id)
```
try to improve it

orest-kostiuk commented on app/controllers/transactions_controller.rb:
```diff
@@ -37,7 +37,7 @@ def destroy
 
   def transaction_params
-    params.require(:transaction).permit(:title, :body, :cost, :account_id, :image, :category_id)
+    params.require(:transaction).permit(:title, :body, :cost, :account_id, :image, :category_id).merge(user_id: current_user.id)
```
try to improve it

orest-kostiuk commented on app/views/accounts/edit.html.erb:
```diff
@@ -1,6 +1,10 @@
+<% if user_signed_in? %>
 <h1>Editing Account</h1>
```
please, add tabs

orest-kostiuk commented on app/views/accounts/index.html.erb:
```diff
@@ -1,30 +1,36 @@
+<% if user_signed_in? %>
+  <h1>Accounts</h1>
 
-<div class="accounts">
```
please, add tabs

orest-kostiuk commented on app/views/accounts/index.html.erb:
```diff
@@ -1,30 +1,36 @@
+        <section class="card-header">
+          <h3><%= link_to account.title, account_path(account), class: 'text-white', style: "text-decoration: none;" %></h3>
         </section>
-        <div style="display: flex;" class="mt-3">
```
please, do not use direct css in html

orest-kostiuk commented on app/views/accounts/index.html.erb:
```diff
@@ -1,30 +1,36 @@
+              <%= truncate strip_tags(account.for_what), length: 150 %>
+            </p>
+          </section>
+          <div style="display: flex;" class="mt-3">
```
please, do not use direct css in html

orest-kostiuk commented on app/views/accounts/new.html.erb:
```diff
@@ -1,5 +1,9 @@
+<% if user_signed_in? %>
 <h1>Creating Account</h1>
```
add tabs

orest-kostiuk commented on app/views/accounts/show.html.erb:
```diff
@@ -1,3 +1,4 @@
+<% if user_signed_in? %>
```
add tabs

orest-kostiuk commented on app/views/categories/edit.html.erb:
```diff
@@ -1,6 +1,10 @@
+<% if user_signed_in? %>
```
add tabs

orest-kostiuk commented on app/views/categories/index.html.erb:
```diff
@@ -1,3 +1,4 @@
+<% if user_signed_in? %>
```
add tabs

orest-kostiuk commented on app/views/categories/new.html.erb:
```diff
@@ -1,5 +1,9 @@
+<% if user_signed_in? %>
```
add tabs

orest-kostiuk commented on app/views/categories/show.html.erb:
```diff
@@ -1,7 +1,11 @@
+<% if user_signed_in? %>
```
add tabs

orest-kostiuk replied:
remove it

orest-kostiuk commented on app/views/transactions/edit.html.erb:
```diff
@@ -1,5 +1,9 @@
+<% if user_signed_in? %>
```
add tabs

orest-kostiuk commented on app/views/transactions/index.html.erb:
```diff
@@ -0,0 +1,4 @@
+<% if user_signed_in? %>
```
remove

orest-kostiuk commented on app/views/transactions/show.html.erb:
```diff
@@ -19,3 +20,6 @@
   <%= link_to 'Back', account_path(@transaction.account_id), class: 'btn btn-dark mt-2' %>
 </div>
+<% else %>
+  <h3><strong>You are not singed in yet.</strong></h3>
```
remove

orest-kostiuk reviewed (commented):
Please, remove from each files check if user signed in
and try to do it by another way

VictorKostiuk committed:
Fixed code review
-.idea added to gitignore;
-changed account and transaction params;
-added tabs
-removed if user signed in from all files.

orest-kostiuk commented on app/controllers/accounts_controller.rb:
```diff
@@ -18,7 +18,7 @@ def show
 
   def create
-    @account = Account.create total_sum_params
+    @account = Account.create(total_sum_params.merge(user_id: current_user))
```
you can improve it more

orest-kostiuk commented on app/controllers/application_controller.rb:
```diff
@@ -1,4 +1,5 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::Base
+  before_action :authenticate_user!
```
👍

orest-kostiuk commented on app/controllers/transactions_controller.rb:
```diff
@@ -5,15 +5,15 @@ class TransactionsController < ApplicationController
 
   def index
-    @transactions = Transaction.all
+    @transactions = Transaction.find
```
fix it

orest-kostiuk commented on app/controllers/transactions_controller.rb:
```diff
@@ -5,15 +5,15 @@ class TransactionsController < ApplicationController
 
   def create
-    @transaction = @account.transactions.build transaction_params
+    @transaction = @account.transactions.build(transaction_params.merge(user_id: current_user))
```
you can improve it more

orest-kostiuk commented on app/views/accounts/index.html.erb:
```diff
@@ -1,30 +1,33 @@
-<h1>Accounts</h1>
+<% if user_signed_in? %>
```
you can improve it more

VictorKostiuk committed:
Fixed code review

VictorKostiuk committed:
Fixed transactions controller

orest-kostiuk reviewed (approved):
looks good👍
